### PR TITLE
Un-pin dependencies.

### PR DESCRIPTION
**Pull request by jmcarp**
Pinning to exact versions can cause conflicts when importing
flask-appbuilder. Pin to minimum versions or version ranges instead.

### CODE REVIEW

**Comment by jmcarp** [comment]
This pulls in some useful updates from @fordguo; sorry I didn't notice your PR earlier! This also drops support for python 3.3, which reached end of life a year ago. Since we're already making breaking changes around flask-login, this seems like a good time to drop python 3.3 as well.

cc @mistercrunch

**Comment by SaurabhBajaj** [comment]
Looks good to me 👍

**Comment by mistercrunch** [comment]
@dpgaspar

**Comment by dpgaspar** [comment]
ok, let me check it

**Comment by SaurabhBajaj** [comment]
@dpgaspar  Can we move this PR forward?

File: flask_appbuilder/models/sqla/__init__.py [hunk 1]
```diff
@@ -30,8 +30,7 @@ class SQLA(SQLAlchemy):
 
         Use it and configure it just like flask_SQLAlchemy
     """
-    def make_declarative_base(self, metadata=None):
-        """Creates the declarative base."""
+    def make_declarative_base(self, model, metadata=None):
```
Comment (dpgaspar) [inline flask_appbuilder/models/sqla/__init__.py]: Can you explain this change?

File: flask_appbuilder/models/sqla/__init__.py [hunk 2]
```diff
@@ -46,7 +45,7 @@ def get_tables_for_bind(self, bind=None):
         return result
 
 
-class ModelDeclarativeMeta(_BoundDeclarativeMeta):
+class ModelDeclarativeMeta(DefaultMeta):
```
Comment (dpgaspar) [inline flask_appbuilder/models/sqla/__init__.py]: Can you explain this change?

**Comment by dpgaspar** [comment]
Thank you @jmcarp , can you explain the changes on SQLA and ModelDeclarativeMeta? is it related to SQLAlchemy version bump?

File: flask_appbuilder/models/sqla/__init__.py (hunk 2 above)
Comment (fordguo) [reply flask_appbuilder/models/sqla/__init__.py]: > _BoundDeclarativeMeta is renamed to DefaultMeta; the old name will be removed in 3.0

see https://github.com/mitsuhiko/flask-sqlalchemy/blob/50944e77522d4aa005fc3c833b5a2042280686d3/CHANGES.rst#version-230
and
mitsuhiko/flask-sqlalchemy#546

File: flask_appbuilder/models/sqla/__init__.py (hunk 1 above)
Comment (jmcarp) [reply flask_appbuilder/models/sqla/__init__.py]: This is the current signature of `make_declarative` base as of flask-sqlachemy 2.3: https://github.com/mitsuhiko/flask-sqlalchemy/blob/master/flask_sqlalchemy/__init__.py#L734

**Comment by jmcarp** [comment]
I think @fordguo and I covered the changes. Is there anything else we should document or explain @dpgaspar ?

**Comment by dpgaspar** [comment]
LGTM, although I don't like it much :P

**Comment by jmcarp** [comment]
@dpgaspar: I pinned all versions to a range like you suggested in https://github.com/dpgaspar/Flask-AppBuilder/pull/814#issuecomment-426546110. Do you think we should use more restricted ranges?

**Comment by dpgaspar** [comment]
No it's good, good work @jmcarp

**Pull request merged**